composables/useBrowserExtension.ts

```diff
@@ -100,18 +102,28 @@ const handleAccountsChanged = async (accounts: string[]) => {
   }
 }
 
+const handleChainChanged = (network: { chainId: string }) => {
+  const { selectedChainId } = storeToRefs(useAppStore())
+
+  selectedChainId.value = network.chainId
+  disconnect()
+  navigateTo(homeRoute())
+}
+
 const handleDisconnect = () => {
   location.reload()
 }
 
-const addProviderEvents = async (provider: any) => {
+const addProviderEvents = async (provider: ProviderAPI) => {
   provider?.on?.('accountsChanged', handleAccountsChanged)
   provider?.on?.('disconnect', handleDisconnect)
+  provider?.on?.('chainChanged', handleChainChanged)
```

Comment:
strange. I did this locally and it was working all. did it require any additional setup?